### DISCUSSION TITLE
Comparison of CUDA-Q with other QC frameworks

### DISCUSSION THREAD

**Discussion by essaathar** [NVIDIA/cuda-quantum#1980] [Q&A]
I wanted to do **runtime comparisons** of various algorithms on CUDA-Q with other quantum computing frameworks like **Qiskit, PennyLane, Cirq, etc.**

1. Would it be reasonable to do so?
2. NVIDIA's **cuQuantum** has been integrated with Qiskit and many other qc platforms (stated in the "Framework Integrations" section [here](https://developer.nvidia.com/cuquantum-sdk)). Since CUDA-Q offers cuQuantum features too (as stated in the "Features" section [here](https://developer.nvidia.com/cuda-q)), is it fair to do the comparison still? I'm confused about how exactly is cuQuantum and if not CUDA-Q, integrated into these other qc software platforms.
3. If it's reasonable to compare, how can I go about doing it? **Note:** I do not have a GPU, I'm running CUDA-Q on colab using the python wheels, hence do not have access to nvidia's qpu and/or gpu targets.

References:
- https://developer.nvidia.com/cuda-q
- https://developer.nvidia.com/cuquantum-sdk

**Comment by efratshabtai** [comment]
CUDA-Q has an optimized integration with cuQuantum to accelerate quantum kernels on GPUs, to achieve best performance. cuQuantum is also integrated with other quantum frameworks to accelerate them on GPUs. 
To get the best CUDA-Q performance, it is recommended to do any benchmarks on Nvidia GPUs.
